### PR TITLE
Add testing of GKE using GCE Subnetworks to Jenkins.

### PR DESCRIPTION
**Pull request by goltermann**
Adding test config.

### CODE REVIEW

**Comment by goltermann** [comment]
@k8s-bot unit test this

**Comment by goltermann** [comment]
@k8s-bot test this

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -472,6 +472,21 @@ case ${JOB_NAME} in
     : ${ENABLE_DEPLOYMENTS:=true}
     ;;
 
+  kubernetes-e2e-gke-subnet)
+    : ${DOGFOOD_GCLOUD:="true"}
+    : ${E2E_CLUSTER_NAME:="jkns-gke-subnet"}
+    : ${E2E_NETWORK:="auto-subnet"}
```
Comment (jlowdermilk) [inline hack/jenkins/e2e.sh]: This network is manually created with custom settings. Should add a comment here stating how to recreate it (ideally with gcloud command to use) if it gets deleted accidentally.

Comment (goltermann) [inline hack/jenkins/e2e.sh]: Done.

Comment (jlowdermilk) [inline hack/jenkins/e2e.sh]: Can you put the gcloud command flags or mention what checkbox(s) to tick on ui? If a month from now it gets deleted, good chance the person fixing the test won't know what options to recreate it with.

Comment (goltermann) [inline hack/jenkins/e2e.sh]: It's straightforward https://cloud.google.com/compute/docs/subnetworks#networks_and_subnetworks : in the CLI or UI, just check the "auto" mode.  No further options.

Comment (goltermann) [inline hack/jenkins/e2e.sh]: Added the gcloud line.

**Comment by goltermann** [comment]
@k8s-bot test this

**Comment by goltermann** [comment]
@k8s-bot test this

**Pull request merged**